Source/Text/GdiFontDefn.cs

```diff
@@ -44,50 +42,43 @@ public IList<RectangleF> MeasureCharacters(ISvgRenderer renderer, string text)
                 format.FormatFlags |= StringFormatFlags.MeasureTrailingSpaces;
-                format.SetMeasurableCharacterRanges((from r in Enumerable.Range(32 * s, numberOfChar)
-                                                     select new CharacterRange(r, 1)).ToArray());
-                regions.AddRange(from r in g.MeasureCharacterRanges(text, _font, new Rectangle(0, 0, numberOfChar * _font.Height, 1000), format)
-                                 select r.GetBounds(g));
+                for (var s = 0; s <= (text.Length - 1) / 32; s++)
+                {
+                    int numberOfChar = Math.Min(32, text.Length - 32 * s);
+                    format.SetMeasurableCharacterRanges((from r in Enumerable.Range(32 * s, numberOfChar)
+                                                         select new CharacterRange(r, 1)).ToArray());
+                    regions.AddRange(from r in g.MeasureCharacterRanges(text, _font, new Rectangle(0, 0, numberOfChar * _font.Height, 1000), format)
+                                     select r.GetBounds(g));
+                }
             }
             return regions;
         }
 
         public SizeF MeasureString(ISvgRenderer renderer, string text)
         {
             var g = GetGraphics(renderer);
-            StringFormat format = StringFormat.GenericTypographic.Clone() as StringFormat;
-            format.SetMeasurableCharacterRanges(new CharacterRange[] { new CharacterRange(0, text.Length) });
-            format.FormatFlags |= StringFormatFlags.MeasureTrailingSpaces;
-            Region[] r = g.MeasureCharacterRanges(text, _font, new Rectangle(0, 0, 1000, 1000), format);
-            RectangleF rect = r[0].GetBounds(g);
+            using (var format = new StringFormat(StringFormat.GenericTypographic))
+            {
+                format.SetMeasurableCharacterRanges(new CharacterRange[] { new CharacterRange(0, text.Length) });
+                format.FormatFlags |= StringFormatFlags.MeasureTrailingSpaces;
+                var r = g.MeasureCharacterRanges(text, _font, new Rectangle(0, 0, 1000, 1000), format);
+                var rect = r[0].GetBounds(g);
 
-            return new SizeF(rect.Width, Ascent(renderer));
+                return new SizeF(rect.Width, Ascent(renderer));
+            }
         }
 
-        private Graphics _graphics;
-        private Graphics GetGraphics(object renderer)
+        private Graphics GetGraphics(ISvgRenderer renderer)
         {
             var provider = renderer as IGraphicsProvider;
             if (provider == null)
             {
-                if (_graphics == null)
-                {
-                    var bmp = new Bitmap(1, 1);
-                    _graphics = Graphics.FromImage(bmp);
-                }
-                return _graphics;
-            }
-            else
-            {
-                return provider.GetGraphics();
+                throw new NotImplementedException("renderer is not IGraphicsProvider");
```

Comment:
Just receive from user without generating in class.

Reply:
I'm not sure about this one - maybe there is a use case for it, it looks like a bug fix.
@erdomke - in case you are around, do you still remember why this (the 1-byte bitmap fallback) was needed?

Reply:
@mrbean-bremen 

I analyzed with code and execution.

This code is performed in the following cases:

1. `renderer` is null.
2. `renderer` does not implement `IGraphicsProvider`.(`renderer` always implements `ISvgRenderer`.)

The first case does not occur with `SvgTextBase` class.(line 304 in SvgTextBase.cs)

The second case occurs with a `renderer` that implements `ISvgRenderer`, such as `DebugRenderer`, but does not implement `IGraphicsProvider`.

Looking at this code, I do not know if implement `ISvgRenderer` but not `IGraphicsProvider`...
In that sense, there is also a way to implement `IGraphicsProvider` with `ISvgRenderer`.

```cs
public interface ISvgRenderer : IGraphicsProvider, IDisposable
```

However, this way forces custom renderer users to make changes.